commit f1915d5162a92d5d5e24aec054417eefea075184
Author: Peter Rosin <peda@axentia.se>
Date:   Thu Mar 29 15:10:54 2018 +0200

    mtd: rawnand: atmel: add module param to avoid using dma
    
    [ Upstream commit efc6362c6f8c1e74b340e2611f1b35e7d557ce7b ]
    
    On a sama5d31 with a Full-HD dual LVDS panel (132MHz pixel clock) NAND
    flash accesses have a tendency to cause display disturbances. Add a
    module param to disable DMA from the NAND controller, since that fixes
    the display problem for me.
    
    Signed-off-by: Peter Rosin <peda@axentia.se>
    Reviewed-by: Boris Brezillon <boris.brezillon@bootlin.com>
    Signed-off-by: Miquel Raynal <miquel.raynal@bootlin.com>
    Signed-off-by: Sasha Levin <alexander.levin@microsoft.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/mtd/nand/raw/atmel/nand-controller.c b/drivers/mtd/nand/raw/atmel/nand-controller.c
index 12f6753d47ae..e686fe73159e 100644
--- a/drivers/mtd/nand/raw/atmel/nand-controller.c
+++ b/drivers/mtd/nand/raw/atmel/nand-controller.c
@@ -129,6 +129,11 @@
 #define DEFAULT_TIMEOUT_MS			1000
 #define MIN_DMA_LEN				128
 
+static bool atmel_nand_avoid_dma __read_mostly;
+
+MODULE_PARM_DESC(avoiddma, "Avoid using DMA");
+module_param_named(avoiddma, atmel_nand_avoid_dma, bool, 0400);
+
 enum atmel_nand_rb_type {
 	ATMEL_NAND_NO_RB,
 	ATMEL_NAND_NATIVE_RB,
@@ -1977,7 +1982,7 @@ static int atmel_nand_controller_init(struct atmel_nand_controller *nc,
 		return ret;
 	}
 
-	if (nc->caps->has_dma) {
+	if (nc->caps->has_dma && !atmel_nand_avoid_dma) {
 		dma_cap_mask_t mask;
 
 		dma_cap_zero(mask);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1149_f1915d5162a92d5d5e24aec054417eefea075184.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1149_f1915d5162a92d5d5e24aec054417eefea075184.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
